apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+| CJS source code using `require()` | CJS: consumers `require()` your package                          | [CJS source and distribution](#cjs-source-and-distribution)                                                                                                  |
+| CJS source code using `require()` | ESM: consumers `import` your package                             | [CJS source and only ESM distribution](#cjs-source-and-only-esm-distribution)                                                                                |
+| CJS source code using `require()` | CJS & ESM: consumers either `require()` or `import` your package | [CJS source and both CJS & ESM distribution](#cjs-source-and-both-cjs-amp-esm-distribution)                                                                  |
+| ESM source code using `import`    | CJS: consumers `require()` your package                          | [ESM source with only CJS distribution](#esm-source-with-only-cjs-distribution)<br />[ESM source and distribution](#esm-source-and-distribution)             |
+| ESM source code using `import`    | ESM: consumers `import` your package                             | [ESM source and distribution](#esm-source-and-distribution)                                                                                                  |
+| ESM: source code uses `import`    | CJS & ESM: consumers either `require()` or `import` your package | [ESM source and both CJS & ESM distribution](#esm-source-and-both-cjs-amp-esm-distribution)<br />[ESM source and distribution](#esm-source-and-distribution) |
+
+### CJS source and distribution
+
+This the "Rum & Coke" of packages: pretty difficult to mess up. You _technially_ may not need _any_ package configuration beyond [`"name"`](https://nodejs.org/api/packages.html#name). But the less arcane, the better: Essentially just declare the package’s exports via the `"exports"` field/field-set.
+
+**Working example**: [cjs-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/cjs-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-and-distribution"
+  // "main": "./index.js"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-and-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.js",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Note that `packageJson.exports["."] = filepath` is shorthand for `packageJson.exports["."].default = filepath`
+
+### CJS source and only ESM distribution
+
+The "Gin & Tonic" of packages: This takes a small bit of finesse but is also pretty straight-forward.
+
+**Working example**: [cjs-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/esm-distro)
```

Comment:
This seems rather uncommon, I am not sure if I have seen any real world packages authored in CJS and then ship as transpiled ESM...

Reply:
I have definitely seen this. DM me if you need specific examples (they are not public).

Reply:
Are there any note-worthy pacakges on npm doing this? Should we even document patterns that can't be found on npm?

Reply:
I don't know if there are any on npm's public registry. I suppose we could tell companies with huge collections of private libraries to figure it out themselves. But why?

Reply:
If a pattern is a popular enough then surely they should have some representation in the public registration to be noticed. Otherwise this may just be some specific hacks that certain companies do, and should be left to their internal documentation. (I know many things that Chinese companies use commonly, e.g. use of "node" in the dependencies to lock Node.js versions, but that's not something I would suggest to document in official tutorials for beginners either).